packages/host/app/components/operator-mode/card-inheritance-panel.gts

```diff
@@ -42,8 +56,42 @@ export default class CardInheritancePanel extends Component<Args> {
     return this.args.adoptionChain?.types;
   }
 
+  @cached
+  get lastModifiedConsumer() {
+    this.args.readyFile.lastModified;
+    this.calculateLastModified();
+    return;
+  }
+
+  private calculateLastModified() {
+    if (this.args.readyFile.lastModified != undefined) {
+      if (
+        Date.now() / 1000 - moment(this.args.readyFile.lastModified).unix() <
+        10
+      ) {
+        this.lastModified = 'Last saved just now';
+      } else {
+        this.lastModified = `Last saved ${moment(
+          this.args.readyFile.lastModified,
+        ).fromNow()}`;
+      }
+    } else {
+      this.lastModified = undefined;
+    }
+  }
+
+  private get fileExtension() {
+    if (!this.args.cardInstance) {
+      return '.' + this.args.readyFile.url.split('.').pop() || '';
+    } else {
+      return '';
+    }
+  }
+
   <template>
     <div class='container' ...attributes>
+      {{! this consumes the last modified date so we can recalcuate it when it invalidates}}
+      {{this.lastModifiedConsumer}}
```

Comment:
This seems vaguely scandalous to me although I can’t make a clearly better solution, could it be a resource somehow?

Reply:
this is because we want this to update as part of a timer *and* in response to the file resource changing, so i added some  consumption such that we get tied into the glimmer invalidation. i _think_ i could rewrite this using a resource. I'll give it a shot.

Reply:
understood, if it doesn’t work out oh well 😀

Reply:
this worked! fixed in https://github.com/cardstack/boxel/pull/663/commits/862abb3cb25494a14d865e65fa151978166aa108

Reply:
nice, this might mean `moment` can be removed? I only used it because it did that kind of `setInterval` thing internally to keep the time updated.